### ISSUE TITLE
[BUG][JAVA][NATIVE] Pojo with Set uses Arraylist, wrong datatype in for loop

### ISSUE THREAD

**Issue by RInverid** [OpenAPITools/openapi-generator#14833]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When using an object with a set, I ran into two separate issues:

1. The generated add to set method assigns ArrayList to the set, which fails compilation
2. The generated toUrlQueryString uses the wrong datatype in a for loop with the set, which fails compilation

##### openapi-generator version

6.4.0

##### OpenAPI declaration file content or url

https://gist.github.com/RInverid/417213fbba82782204ddce87b22d5d62

##### Generation Details

Java, native http library

##### Steps to reproduce
Try to build the generated client, it fails to build due assigning a list to a set, and looping over the enum instead of an array

##### Suggest a fix

Issue 1:
variable is correctly assigned [here](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache#L85) and incorrectly assigned [here](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache#L146). Caused by commit https://github.com/OpenAPITools/openapi-generator/commit/fd45b74128c3eea9fcea0172095f5b369d7891e7

Issue 2:
Datatype is incorrect [here](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache#L389), in this case it should be an enum.

**Comment by mvanglabbeek** [comment]
I experience the same issue. Version 6.2.1 initialized the `java.util.Set` with `this.fieldName = new LinkedHashSet<>();`, which is correct. Version 6.3.0 and version 6.4.0 initializes the `java.util.Set` with `this.fieldName = new ArrayList<>();` (without `importing java.util.ArrayList`), which is broken code.

**Comment by RInverid** [comment]
@wing328 the official maintainer of Java (Java 11 Native HTTP client) @bbdouglas, has been inactive since 2019. Maybe this API client should be marked as unsupported?

**Comment by wing328** [comment]
I've filed https://github.com/OpenAPITools/openapi-generator/pull/14961 to fix the issue. Please pull it to give it a try. You may need the option `--additional-properties supportUrlQuery=false` to skip the `toUrlQuery` method, e.g.

```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g java --library native -i https://gist.githubusercontent.com/RInverid/417213fbba82782204ddce87b22d5d62/raw/d85d89ec41db9e6a93400426ffc8178b5879305b/spec.json -o /tmp/native --additional-properties supportUrlQuery=false
```

**Comment by wing328** [comment]
> Maybe this API client should be marked as unsupported?

The `native` library support is maintained by the community. There has been a lot of enhancements so I don't think we should deprecate it.

We appreciate @bbdouglas for contributing the first version of `native` library support

**Comment by wing328** [comment]
the fix has been merged and will be included in the upcoming 6.5.0 release
